### PR TITLE
[IMP] account,sale,purchase,l10n_it_edi: refactor tax details management for rounding

### PR DESCRIPTION
**Pull request by antonrom1**
Before `_add_tax_details_in_base_line` populated tax details like `tax_amount_currency` directly, and `_round_base_lines_tax_details` subsequently moved these values to new keys prefixed with `raw_` while rounding the originals. The problem with that is the meaning of certain dictionary keys like `tax_amount_currency` changed depending on the method that processed them

Now, `_add_tax_details_in_base_line` only adds raw values (e.g., `raw_tax_amount_currency`), and `_round_base_lines_tax_details` separately adds the rounded values

This simplifies state management and makes sure that raw and rounded values are clearly separated.

related pr: https://github.com/odoo/enterprise/pull/74121

task: 4277573

### CODE REVIEW

File: addons/account/models/account_tax.py [hunk 1]
```diff
@@ -1635,9 +1627,10 @@ def _add_accounting_data_to_base_line_tax_details(self, base_line, company, incl
             }
             tax_reps_data = tax_data['tax_reps_data'] = []
             for tax_rep in tax_reps:
+                tax_amount_currency = tax_data.get('tax_amount_currency') or tax_data.get('raw_tax_amount_currency')
```
Comment (antonrom1) [inline addons/account/models/account_tax.py:1630]: It's a bit crappy, but if we want to keep current behaviour without doing more refactoring I don't see a better solution

For reference `AccountTax.compute_all` calls `_add_accounting_data_to_base_line_tax_details` before adding rounded tax details. And everywhere else `_add_accounting_data_in_base_lines_tax_details` is called with base lines with rounded data.

File: addons/account/models/account_tax.py [hunk 2]
```diff
@@ -1635,9 +1627,10 @@ def _add_accounting_data_to_base_line_tax_details(self, base_line, company, incl
             }
             tax_reps_data = tax_data['tax_reps_data'] = []
             for tax_rep in tax_reps:
+                tax_amount_currency = tax_data.get('tax_amount_currency', tax_data.get('raw_tax_amount_currency'))
```
Comment (antonrom1) [inline addons/account/models/account_tax.py:1630]: I don't like this but `_add_accounting_data_to_base_line_tax_details` is being called with base_lines in a rounded and non-rounded state, so without further refactoring it's the only way I see to keep current behaviour

For reference `AccountTax.compute_all` calls `_add_accounting_data_to_base_line_tax_details` before adding rounded tax details. And everywhere else `_add_accounting_data_in_base_lines_tax_details` is called with base lines with rounded data.

File: addons/account/models/account_tax.py [hunk 3]
```diff
@@ -1650,8 +1643,11 @@ def _add_accounting_data_to_base_line_tax_details(self, base_line, company, incl
                 tax_reps_data,
                 key=lambda tax_rep: (-abs(tax_rep['tax_amount_currency']), -abs(tax_rep['tax_amount'])),
             )
-            for field, field_currency in (('tax_amount_currency', currency), ('tax_amount', company_currency)):
-                total_error = tax_data[field] - total_tax_rep_amounts[field]
+            for field, field_raw, field_currency in (
+                    ('tax_amount_currency', 'raw_tax_amount_currency', currency),
+                    ('tax_amount', 'raw_tax_amount', company_currency),
+            ):
+                total_error = tax_data.get(field, tax_data.get(field_raw)) - total_tax_rep_amounts[field]
```
Comment (antonrom1) [inline addons/account/models/account_tax.py:1650]: same here

File: addons/account/models/account_tax.py [hunk 4]
```diff
@@ -1328,15 +1328,15 @@ def _add_tax_details_in_base_line(self, base_line, company, rounding_method=None
         )
         rate = base_line['rate']
         tax_details = base_line['tax_details'] = {
-            'total_excluded_currency': taxes_computation['total_excluded'],
-            'total_excluded': taxes_computation['total_excluded'] / rate if rate else 0.0,
-            'total_included_currency': taxes_computation['total_included'],
-            'total_included': taxes_computation['total_included'] / rate if rate else 0.0,
+            'raw_total_excluded_currency': taxes_computation['total_excluded'],
```
Comment (smetl) [inline addons/account/models/account_tax.py:1331]: can you update the docstring as well plz?

File: addons/account/models/account_tax.py [hunk 5]
```diff
@@ -1650,8 +1647,14 @@ def _add_accounting_data_to_base_line_tax_details(self, base_line, company, incl
                 tax_reps_data,
                 key=lambda tax_rep: (-abs(tax_rep['tax_amount_currency']), -abs(tax_rep['tax_amount'])),
             )
-            for field, field_currency in (('tax_amount_currency', currency), ('tax_amount', company_currency)):
-                total_error = tax_data[field] - total_tax_rep_amounts[field]
+            for field, field_currency in (
+                    ('tax_amount_currency', currency),
+                    ('tax_amount', company_currency),
+            ):
```
Comment (smetl) [inline addons/account/models/account_tax.py:1653]:
```suggestion
            for field, field_currency in (
                ('tax_amount_currency', currency),
                ('tax_amount', company_currency),
            ):
```

File: addons/account/models/account_tax.py [hunk 6]
```diff
@@ -1650,8 +1647,14 @@ def _add_accounting_data_to_base_line_tax_details(self, base_line, company, incl
...
+                tax_amount = tax_data.get(field)
+                if self.env.context.get('compute_all_use_raw_base_lines'):
+                    tax_amount = tax_data.get("raw_" + field)
```
Comment (smetl) [inline addons/account/models/account_tax.py:1656]:
```suggestion
                    tax_amount = tax_data.get(f"raw_{field}")
```

**Review by smetl — APPROVED** [review]
@antonrom1 Can you rebase plz ?

**Review by smetl — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**